### PR TITLE
add OptionType kInt32T and kInt64T

### PR DESCRIPTION
**Pull request by miasantreble**

### CODE REVIEW

File: util/string_util.cc [hunk 1]
```diff
@@ -303,6 +312,31 @@ uint64_t ParseUint64(const std::string& value) {
   return num;
 }
 
+int64_t ParseInt64(const std::string& value) {
+  size_t endchar;
+#ifndef CYGWIN
+  int64_t num = std::stoull(value.c_str(), &endchar);
+#else
+  char* endptr;
+  int64_t num = std::strtoul(value.c_str(), &endptr, 0);
```
Comment (sagar0) [inline util/string_util.cc]: I believe you should be using `std::strtol` / `std::strtoll` here and line 318.  
Looks good otherwise.

Comment (miasantreble) [reply util/string_util.cc]: good catch!

**Comment by miasantreble** [comment]
I was planning to use it in https://github.com/facebook/rocksdb/pull/5022 for `rocksdb_options_set_max_write_buffer_size_to_maintain` because it might exceed MAX_INT and we also want to reserve negative values for a special case

File: util/string_util.cc [hunk 2]
```diff
@@ -303,6 +312,31 @@ uint64_t ParseUint64(const std::string& value) {
...
+  if (endchar < value.length()) {
+    char c = value[endchar];
+    if (c == 'k' || c == 'K')
+      num <<= 10LL;
+    else if (c == 'm' || c == 'M')
+      num <<= 20LL;
+    else if (c == 'g' || c == 'G')
+      num <<= 30LL;
+    else if (c == 't' || c == 'T')
+      num <<= 40LL;
```
Comment (sagar0) [inline util/string_util.cc]: I am not fully sure, but I believe left shifting a negative integer could be undefined. If so, we might need to guard agains such an input.

Comment (miasantreble) [reply util/string_util.cc]: test coverage added to add more confidence

**Pull request closed without merging**